### PR TITLE
Log missing/unexpected keys on non-strict checkpoint loads

### PR DESCRIPTION
**Pull request by EHxuban11**
## Summary

Makes non-strict checkpoint loading loud. Families with `_strict_loading() == False` (YOLOX) previously discarded missing/unexpected state-dict keys without any trace: a partially matching checkpoint would "load" and then quietly predict with fresh-initialized tensors wherever keys were absent.

Shape mismatches always raise (PyTorch raises them regardless of `strict`), so the silent failure mode is specifically *named* key mismatches. During the RF100-VL yolox-nano investigation this cost hours: the load path had to be manually instrumented before it could be ruled out as the cause of a 0.57→0.16 metric collapse (the actual cause was the BN eps bug fixed in #700 — but a loud loader would have excluded this hypothesis in seconds).

### Behavior

- Both non-strict load sites (`BaseModel.__init__` dict path and `_load_weights`) route through a new `_load_state_dict_logged` helper.
- On any missing/unexpected keys: one WARNING with counts and the first five names of each class, plus full key lists at DEBUG.
- Healthy loads stay completely silent; strict families are unaffected (they raise before the helper's check matters).

Note: legitimate partial loads (e.g. permitted cross-task loads where the head is intentionally re-initialized) now emit this warning too — that is by design, since "your head weights are fresh-initialized" is exactly what a user should see.

### Tests

`tests/unit/test_nonstrict_load_logging.py`: healthy load is silent; a state dict with 3 deleted head keys and 1 bogus key produces exactly one warning naming both counts and example keys.

## Test plan

- [x] New unit tests
- [x] Full `pytest tests/unit`: 4460 passed, 169 skipped, 1 xfailed, 0 failed
- [ ] CI

<details><summary><h3>Greptile Summary</h3></summary>

The PR adds centralized warning and debug logging for state-dict key mismatches and routes the base and YOLO-NAS loading paths through it.
- Adds `_load_state_dict_logged` to report missing and unexpected keys.
- Adds unit coverage for healthy and mismatched YOLOX state dictionaries.
- Documents the behavior in the changelog.
</details>

<details><summary><h3>Confidence Score: 4/5</h3></summary>

The PR is not yet safe to merge because several custom non-strict checkpoint loaders still silently accept key mismatches.

The shared and YOLO-NAS paths now report mismatches, but the previously reported bypass remains in custom loaders such as D-FINE, DEIM, EC, RF-DETR, PIDNet, and DINOv2, where one or both returned mismatch lists are not surfaced.

**Files Needing Attention:** libreyolo/models/base/model.py and the affected custom model loaders under libreyolo/models/
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| libreyolo/models/base/model.py | Adds the shared state-dict loading helper and uses it for dictionary and standard file-loading paths. |
| libreyolo/models/yolonas/model.py | Routes YOLO-NAS checkpoint loading through the shared mismatch logger. |
| tests/unit/test_nonstrict_load_logging.py | Tests silent healthy loading and warning details for missing and unexpected YOLOX keys. |
| CHANGELOG.md | Documents the new non-strict checkpoint mismatch diagnostics. |

</details>

<sub>Reviews (2): Last reviewed commit: ["Route YOLO-NAS custom loader through log..."](https://github.com/libreyolo/libreyolo/commit/dc4c696cb27567e8e417021dc32985eb7e224666) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49686489)</sub>

### CODE REVIEW

File: libreyolo/models/base/model.py [hunk 1]
```diff
@@ -817,13 +817,43 @@ def _load_weights(self, model_path: str):
                 apply_quant_structure(self, quant_manifest)
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/models/base/model.py:817]: Fair point, but most custom loaders already police the result with family-specific rules: D-FINE/DEIM/EC capture missing/unexpected and raise on unexpected keys (tolerating only regenerated anchor buffers), and DINOv2/RF-DETR/PIDNet validate key sets against the expected architecture. Routing those through this generic helper would replace targeted errors with generic warnings, and would warn on every healthy load for families with by-design drift (L2CS's vestigial fc_finetune, RTMDet's share_conv aliasing). The one custom loader that was genuinely blind was YOLO-NAS - it discarded the load result entirely. Fixed in dc4c696c: it now routes through _load_state_dict_logged, and the helper docstring documents the contract for custom loaders.
